autopush/main.py

```diff
@@ -345,9 +345,17 @@ def setup(self, rotate_tables=True):
 
     def run(self):
         try:
-            self.push_server.run()
+            self.push_server.start()
+            while True:
+                try:
+                    # handle a graceful shutdown on SIGINT w/ a busy
+                    # loop. we can't Thread.join because SIGINT won't
+                    # interrupt it
+                    time.sleep(6000)
+                except KeyboardInterrupt:
```

Comment:
why not signal? (it'd handle other kills too, right?)

Reply:
signal could special handle potentially any kill but you have to register every specific signal. I could have used a signal handler but I'd actually still need this busy loop to keep the main thread going (python signal handlers should only run on the main thread -- and I even just double checked, they wouldn't have run if the main thread has died)

the special handling for KeyboardInterrupt here isn't even totally necessary, it's just to hide the traceback on the console for the common ctrl-c case